src/Magento/FunctionalTestingFramework/Test/Util/TestObjectExtractor.php

```diff
@@ -155,4 +170,34 @@ public function extractTestData($testData)
             throw new XmlException($exception->getMessage() . ' in Test ' . $filename);
         }
     }
+
+    /**
+     * Append names of test files, including merge files, in description annotation
+     *
+     * @param string $description
+     * @param array  $fileNames
+     *
+     * @return string
+     */
+    private function appendFileNamesInDescriptionAnnotation($description, $fileNames)
+    {
+        $title = '';
+        foreach ($fileNames as $fileName) {
+            if (!empty($fileName && realpath($fileName) !== false)) {
+                $fileName = realpath($fileName);
+                $relativeFileName = ltrim(
+                    str_replace(MAGENTO_BP, '', $fileName),
+                    DIRECTORY_SEPARATOR
+                );
+                if (!empty($relativeFileName)) {
+                    if (empty($title)) {
```

Comment:
You can get rid of this conditional by setting `$title = '<br><br>Test Files:<br>'` on line 184 and returning `return $description .= $title` instead.

I understand that this would mean if no filenames are passed in that you'd have a partial description, but I think that's okay since we want to be able to easily tell processing of it went wrong.

Reply:
Another side effect of this is that it will break unit tests.